src/ripple/overlay/impl/PeerImp.cpp

```diff
@@ -1080,8 +1080,7 @@ PeerImp::onMessage (std::shared_ptr <protocol::TMProposeSet> const& m)
 {
     protocol::TMProposeSet& set = *m;
 
-    if (overlay_.setup().expire &&
-            set.has_hops() && ! slot_->cluster())
+    if (set.has_hops() && ! slot_->cluster())
```

Comment:
@scottschurr Fair point. I discussed it with Vinnie offline, so I had a bit more background. `expire` is a flag in the config file that indicates whether you want your proposals to expire. It does not appear to be documented, but that may be because it's experimental. Still, it probably should be somewhere in here. https://github.com/vinniefalco/rippled/blob/fix-hops/doc/rippled-example.cfg#L362-L392.

More important than what `expire` does, though, is that it's intention is to control the behavior of the originator, not the relayer. If the node that receives the message knows how to handle the `hops` field, it should increment it before relaying it, regardless of local configuration. (And, BTW, the originator ensures that it only sends the messages to nodes that know how to handle `hops`. https://github.com/ripple/rippled/blob/develop/src/ripple/overlay/impl/OverlayImpl.cpp#L725)